Dockerfile

```diff
@@ -1,101 +1,33 @@
-# ==
-# builder
-# ==
+FROM node:16-alpine
 
-# application builder
-FROM node:16 AS builder
+ARG PNPM_VERSION
 
-WORKDIR /usr/app
+RUN npm install -g pnpm@${PNPM_VERSION}
 
-ARG API_URL
+USER node
 
-# Install pnpm
-RUN npm install -g pnpm
+WORKDIR /home/node/app
 
-# copy package.json and package-lock.json files
-COPY package.json .
 COPY pnpm-lock.yaml .
-COPY .npmrc .
```

Comment:
Isn't it necessary to keep `.npmrc` and to copy `.pnpmfile.cjs` here?

Reply:
It's not! `pnpm fetch` only populates the pnpm global cache of packages based on the package lock, but does not do anything with the `node_modules` file that `pnpm install` creates. Only `pnpm install` uses `.npmrc` to change installation settings and `.pnpmfile.cjs` to modify resolution (which is documented in the `pnpm-lock.yaml` so there is no risk of `.pnpmfile.cjs` creating a resolution to a package that was not included in the lockfile already and therefore didn't exist in the store).

Reply:
Nevermind, I read that `fetch` does require only the lockfile 👍

Reply:
Ups, I hadn't read your reply before posting my last message, thanks for the explanation! 😄